security/src/main/java/org/frankframework/lifecycle/servlets/OAuth2Authenticator.java

```diff
@@ -106,39 +109,42 @@ public SecurityFilterChain configure(HttpSecurity http) throws Exception {
 		http.oauth2Login(login -> login
 				.clientRegistrationRepository(clientRepository) // Explicitly set, but can also be implicitly implied.
 				.authorizedClientService(new InMemoryOAuth2AuthorizedClientService(clientRepository))
-				.failureUrl(oauthBaseUrl + "/oauth2/failure/")
-				.authorizationEndpoint(endpoint -> endpoint.baseUri(oauthBaseUrl + OAuth2AuthorizationRequestRedirectFilter.DEFAULT_AUTHORIZATION_REQUEST_BASE_URI))
+				.failureUrl(servletPath + "/oauth2/failure/")
+				.authorizationEndpoint(endpoint -> endpoint.baseUri(servletPath + OAuth2AuthorizationRequestRedirectFilter.DEFAULT_AUTHORIZATION_REQUEST_BASE_URI))
 				.userInfoEndpoint(endpoint -> endpoint.userAuthoritiesMapper(authorityMapper))
-				.loginProcessingUrl(oauthBaseUrl + "/oauth2/code/*"));
+				.loginProcessingUrl(servletPath + "/oauth2/code/*"));
 
 		return http.build();
 	}
 
 	private void configure() throws FileNotFoundException {
-		if(StringUtils.isEmpty(clientId) || StringUtils.isEmpty(clientSecret)) {
+		if (StringUtils.isEmpty(clientId) || StringUtils.isEmpty(clientSecret)) {
 			throw new IllegalStateException("clientId and clientSecret must be set");
 		}
+		if (StringUtils.isEmpty(provider)) {
+			throw new IllegalStateException("A provider must be set");
+		}
 
 		roleMappingURL = ClassUtils.getResourceURL(roleMappingFile);
 		if(roleMappingURL == null) {
 			throw new FileNotFoundException("unable to find OAUTH role-mapping file ["+roleMappingFile+"]");
 		}
 		log.info("found rolemapping file [{}]", roleMappingURL);
 
-		oauthBaseUrl = computeBaseUrl();
-		log.debug("using oauth baseurl [{}]", oauthBaseUrl);
+		String baseUrl = computeBaseUrl();
```

Comment:
Omdat er ook een instance field `baseUrl` is, is de code denk ik duidelijker als je de lokale variabele hier kan hernoemen naar bv `computedBaseUrl` of zo.